### PR TITLE
[Feature] Render max of Four computed rates per day

### PR DESCRIPTION
**Pull request by ghost**
#### What does this PR do?
- Ensures that a maximum of four rates are visible every day on the HistoricRates page.
- Ensures that the rates follow the `USD` `GBP` `EUR`, `YEN` order strictly.

#### Description of Task to be completed?
- Add functions to size data to acceptable level.
- Add `periodOfTheDay` to computedRates query.
- Pass data through resizing function before render.
- Render `NoData` component when computed rates has no data.
- Add extra tests

### CODE REVIEW

File: src/dataApi/index.js [hunk 1]
```diff
@@ -70,3 +70,27 @@ export const sortTodayRates = (data) => {
...
+export const stripCurrencyToOneInstance = (structuredRates) => {
+  const USDItem = structuredRates.find((rate) => rate.currency === 'USD');
+  const GBPItem = structuredRates.find((rate) => rate.currency === 'GBP');
+  const EURItem = structuredRates.find((rate) => rate.currency === 'EUR');
+  const YENItem = structuredRates.find((rate) => rate.currency === 'YEN');
+  const allCurrencyRates = [USDItem, GBPItem, EURItem, YENItem];
+  return allCurrencyRates.filter(Boolean);
+};
+
+export const trimRates = (ratesArray) => {
```
Comment (omorilewa) [inline src/dataApi/index.js:91]: Elegant, I must say.

Comment (ghost) [reply src/dataApi/index.js:91]: Wry smile. Merci!

**Comment by omorilewa** [comment]
@oreoluwade I noticed the Historic Rates Body does not scroll. Kindly look into this. I added spaces, you can pull and see it does not scroll.

**Comment by omorilewa** [comment]
Also, @oreoluwade When there is a network error e.g If the internet is not going through, the view breaks and I get the error shown below
screen shot 2018-06-10 at 5 44 01 pm

Can we make provision to manage this or at least prevent the error.

**Comment by ghost** [comment]
I have implemented neither a proper scroll nor pagination; those are features definitely going to be implemented.

File: src/components/HistoricRatesItem.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ export default class HistoricRatesItem extends Component {
 
   render() {
     const { ratesPerDate, date } = this.state;
+    const sizedRates = trimRates(ratesPerDate);
```
Comment (omorilewa) [inline src/components/HistoricRatesItem.js]: Is it possible to use a different variable name.

Comment (ghost) [reply src/components/HistoricRatesItem.js]: You'd have to be specific. A different variable name for which?

Comment (omorilewa) [reply src/components/HistoricRatesItem.js]: For `sizedRates`.

Comment (ghost) [reply src/components/HistoricRatesItem.js]: Okay.

**Review by omorilewa — DISMISSED** [review]

**Review by omorilewa — APPROVED** [review]

**Pull request merged**